api/provider/aws/deploy_create.go

```diff
@@ -47,6 +47,24 @@ func (d *DeployProvider) createTaskDefinition(taskDefinition *ecs.TaskDefinition
 		input.SetNetworkMode(nm)
 	}
 
+	// Because of the logic in renderTaskDefinition(), taskDefinition.RequiresCompatibilities
+	// should always contain one or both of "EC2" and "FARGATE"
+	input.SetRequiresCompatibilities(taskDefinition.RequiresCompatibilities)
+
+	if cpu := aws.StringValue(taskDefinition.Cpu); cpu != "" {
```

Comment:
this (and memory below) is essentially saying "if thing is set, then set thing"; it's redundant.

Reply:
✅

Reply:
@zpatrick I wonder if this is actually desired behavior. Do we expect the following two task definitions, once they're registered, to be functionally identical?
```
{
  ContainerDefinitions: [{ . . . }],
  RequiresCompatibilities: [ "EC2"],
}
```
vs
```
{
  ContainerDefinitions: [{ . . . }],
  Cpu: "",
  Memory: "",
  RequiresCompatibilities: [ "EC2" ],
}
```

Reply:
as sorry, didn't see that `input` and `taskDefinition` were different variables

Reply:
✅